utils/p4runtime_lib/convert.py

```diff
@@ -48,7 +48,17 @@ def bitwidthToBytes(bitwidth):
 
 def encodeNum(number, bitwidth):
     byte_len = bitwidthToBytes(bitwidth)
+    # If number is negative, calculate the positive number that its
+    # 2's complement encoding would look like in 'bitwidth' bits.
+    orig_number = number
+    if number < 0:
+        if number < -(2 ** (bitwidth-1)):
+            raise Exception("Negative number, %d, has 2's complement representation that does not fit in %d bits" % (number, bitwidth))
+        number = (2 ** bitwidth) + number
     num_str = '%x' % number
+    if orig_number < 0:
+        print("CONVERT_NEGATIVE_NUMBER debug: orig_number=%s number=%s bitwidth=%d num_str='%s'"
+              "" % (orig_number, number, bitwidth, num_str))
```

Comment:
If all of the above changes look good, this last 'if orig_number < 0:' code is just extra debug code that I used while testing these changes.  I can remove them from the PR in a later commit if this all seems to work as desired.

Note that there is no change in behavior to the version of the code without this PR when only non-negative action parameter values are used.